app/scripts/modules/core/src/pipeline/config/triggers/artifacts/base64/base64.artifact.ts

```diff
@@ -0,0 +1,87 @@
+import { module } from 'angular';
+
+import { PIPELINE_CONFIG_PROVIDER } from 'core/pipeline/config/pipelineConfigProvider';
+import { IArtifact } from 'core/domain/IArtifact';
+import { PipelineConfigProvider } from 'core/pipeline';
+
+import './base64.artifact.less';
+
+export const BASE64_ARTIFACT = 'spinnaker.core.pipeline.trigger.github.artifact';
+module(BASE64_ARTIFACT, [
+  PIPELINE_CONFIG_PROVIDER,
+]).config((pipelineConfigProvider: PipelineConfigProvider) => {
+  pipelineConfigProvider.registerArtifactKind({
+    label: 'Base64',
+    description: 'An embedded base64 blob.',
+    key: 'base64',
+    isDefault: true,
+    isMatch: false,
+    controller(artifact: IArtifact) {
+      'ngInject';
+      this.artifact = artifact;
+      this.artifact.type = 'embedded/base64';
+      this.decoded = '';
+      this.encodeDecodeError = null;
+
+      if (this.artifact.reference) {
+        try {
+          this.decoded = atob(this.artifact.reference);
+        } catch (decodeError) {
+          this.encodeDecodeError = decodeError;
+        }
+      }
+
+      this.onContentChange = () => {
+        this.encodeDecodeError = null;
+        try {
+          this.artifact.reference = btoa(this.decoded);
+        } catch (encodeError) {
+          this.encodeDecodeError = encodeError;
+        }
+      };
+    },
+    controllerAs: 'ctrl',
+    template: `
+<div class="col-md-12">
+  <div class="form-group row">
+    <label class="col-md-2 sm-label-right">
+      Name
+    </label>
+    <div class="col-md-8">
+      <input type="text"
+             placeholder="base64-artifact"
+             class="form-control input-sm"
+             ng-model="ctrl.artifact.name" />
+    </div>
+  </div>
+  <div ng-if="ctrl.encodeDecodeError != null" class="form-group row">
```

Comment:
In most of the spots where we surface this kind of error (see the Pipeline JSON edit modal), we put the error below the text box.